lib/rules/no-loss-of-precision.js

```diff
@@ -0,0 +1,196 @@
+         */
+        function addDecimalPointToNumber(stringNumber) {
+            return `${stringNumber.slice(0, 1)}.${stringNumber.slice(1)}`;
+        }
+
+        /**
+         * Returns the number stripped of leading zeros
+         * @param {string} numberAsString the string representation of the number
+         * @returns {string} the stripped string
+         */
+        function removeLeadingZeros(numberAsString) {
+            return numberAsString.replace(/^0*/u, "");
+        }
+
+        /**
+         * Returns the number stripped of trailing zeros
+         * @param {string} numberAsString the string representation of the number
+         * @returns {string} the stripped string
+         */
+        function removeTrailingZeros(numberAsString) {
+            return numberAsString.replace(/0*$/u, "");
+        }
+
+        /**
+         * Converts an integer to to an object containing the the integer's coefficient and order of magnitude
+         * @param {string} stringInteger the string representation of the integer being converted
+         * @returns {Object} the object containing the the integer's coefficient and order of magnitude
+         */
+        function normalizeInteger(stringInteger) {
+            const significantDigits = removeTrailingZeros(stringInteger);
+
+            return {
+                magnitude: stringInteger.length - 1,
+                coefficient: addDecimalPointToNumber(significantDigits)
+            };
+        }
+
+        /**
+         *
+         * Converts a float to to an object containing the the floats's coefficient and order of magnitude
+         * @param {string} stringFloat the string representation of the floate being converted
+         * @returns {Object} the object containing the the integer's coefficient and order of magnitude
+         */
+        function normalizeFloat(stringFloat) {
+            if (stringFloat.startsWith("0") || stringFloat.startsWith(".")) {
+                const decimalDigits = stringFloat.split(".").pop();
+                const significantDigits = removeLeadingZeros(decimalDigits);
+
+                return {
+                    magnitude: significantDigits.length - decimalDigits.length - 1,
+                    coefficient: addDecimalPointToNumber(significantDigits)
+                };
+
+            }
+            return {
+                magnitude: stringFloat.indexOf(".") - 1,
+                coefficient: addDecimalPointToNumber(stringFloat.replace(".", ""))
+
+            };
+        }
+
+
+        /**
+         * Converts a base ten number to proper scientific notation
+         * @param {string} stringNumber the string representation of the base ten number to be converted
+         * @returns {string} the number converted to scientific notation
+         */
+        function convertNumberToScientificNotation(stringNumber) {
+            const absoluteValue = stringNumber.startsWith("-") ? stringNumber.replace("-", "") : stringNumber;
```

Comment:
I don't think this condition will ever be true, because JS doesn't actually have any negative number literals -- it just has a unary `-` operator that gets applied to positive number literals.